--- Makefile.orig	2018-02-14 12:00:51.134484423 +0100
+++ Makefile	2018-02-14 12:00:27.189838963 +0100
@@ -211,6 +211,10 @@
 	cp $(BUILD_DIR)/ELSI/*.mod $(INC_DIR)
 	cp $(BUILD_DIR)/ELSI/*.h $(INC_DIR)
 	cp $(BUILD_DIR)/ELSI/*.a $(LIB_DIR)
+	mkdir -p $(DESTDIR)/$(prefix)/lib
+	mkdir -p $(DESTDIR)/$(prefix)/include
+	cp $(LIB_DIR)/*.a $(DESTDIR)/$(prefix)/lib
+	cp $(INC_DIR)/*.mod $(DESTDIR)/$(prefix)/include
 	@echo ================================
 	@echo = ELSI installed successfully. =
 	@echo ================================
