diff -urN '--exclude=.git' '--exclude=aotus' '--exclude=.arch.make' '--exclude=.setup.make' '--exclude=README.md' '--exclude=quick_test.sh' flook-0.7.0.old/Makefile flook.new/Makefile
--- flook-0.7.0.old/Makefile	2017-02-28 08:40:09.000000000 +0100
+++ flook.new/Makefile	2018-02-09 14:08:48.694304298 +0100
@@ -25,6 +25,12 @@
 
 # SMEKAENDSETTINGS (DO NOT DELETE)
 
+ifdef PLATFORM
+_LUA_PLATFORM := $(PLATFORM)
+else
+_LUA_PLATFORM := linux
+endif
+
 
 # Define internal aotus library build
 .PHONY: prep-aotus lib-aotus clean-aotus
@@ -32,11 +38,13 @@
 ifdef SETUP
 	(mkdir -p aotus/obj ; cd aotus/obj ; \
 		echo "TOP_DIR =../../$(TOP_DIR)/aotus" > Makefile ; \
+		echo "PLATFORM = $(_LUA_PLATFORM)" >> Makefile ; \
 		echo "include ../../$(TOP_DIR)/aotus/Makefile" >> Makefile )
 
 else
 	(mkdir -p aotus/obj ; cd aotus/obj ; \
 		echo "TOP_DIR =../../$(TOP_DIR)/aotus" > Makefile ; \
+		echo "PLATFORM = $(_LUA_PLATFORM)" >> Makefile ; \
 		echo "include ../../$(TOP_DIR)/aotus/Makefile" >> Makefile )
 endif
 lib-aotus: prep-aotus
diff -urN '--exclude=.git' '--exclude=aotus' '--exclude=.arch.make' '--exclude=.setup.make' '--exclude=README.md' '--exclude=quick_test.sh' flook-0.7.0.old/Makefile.project flook.new/Makefile.project
--- flook-0.7.0.old/Makefile.project	2017-02-28 08:40:09.000000000 +0100
+++ flook.new/Makefile.project	2018-02-09 14:10:31.104766671 +0100
@@ -60,7 +60,7 @@
 ifneq ($(LIBRARIES),)
 install-mod:
 # This should only pass if one wishes to install the flook library as well
-	install -pm $(_OCT_rwrr) -t $(FMOD_DIR) *.mod
+	install -pm $(_OCT_rwrr) -t $(DESTDIR)$(FMOD_DIR) *.mod
 endif
 
 
diff -urN '--exclude=.git' '--exclude=aotus' '--exclude=.arch.make' '--exclude=.setup.make' '--exclude=README.md' '--exclude=quick_test.sh' flook-0.7.0.old/smeka/Makefile.directory flook.new/smeka/Makefile.directory
--- flook-0.7.0.old/smeka/Makefile.directory	2017-02-28 08:40:09.000000000 +0100
+++ flook.new/smeka/Makefile.directory	2018-02-09 14:46:31.914704395 +0100
@@ -16,30 +16,30 @@
 
 
 # Default directories
-PREFIX ?= $(HOME)/$(PROJECT_NAME)
+prefix ?= $(HOME)/$(PROJECT_NAME)
 # Include directory (.h)
-INC_DIR ?= $(PREFIX)/include
+INC_DIR ?= $(prefix)/include
 # Module directory (.mod)
 FMOD_DIR ?= $(INC_DIR)
 # Executables
-BIN_DIR ?= $(PREFIX)/bin
+BIN_DIR ?= $(prefix)/bin
 # Library directory (.a/.so)
-LIB_DIR ?= $(PREFIX)/lib
+LIB_DIR ?= $(prefix)/lib
 # shared folder
-SHARE_DIR ?= $(PREFIX)/share
+SHARE_DIR ?= $(prefix)/share
 # Manual folder (man ***)
 MAN_DIR ?= $(SHARE_DIR)/man
 # pkg-config files
 PKG_DIR ?= $(LIB_DIR)/pkgconfig
 # additional files (settings etc.)
-ETC_DIR ?= $(PREFIX)/etc
+ETC_DIR ?= $(prefix)/etc
 
 
 # Print out settings from this file
 .PHONY: smeka-settings-dir
 smeka-settings-dir:
 	@echo "Directory settings:"
-	@echo "Prefix            (PREFIX) : $(PREFIX)"
+	@echo "Prefix            (prefix) : $(prefix)"
 	@echo "Includes         (INC_DIR) : $(INC_DIR)"
 	@echo "Fortran modules (FMOD_DIR) : $(FMOD_DIR)"
 	@echo "Binaries         (BIN_DIR) : $(BIN_DIR)"
diff -urN '--exclude=.git' '--exclude=aotus' '--exclude=.arch.make' '--exclude=.setup.make' '--exclude=README.md' '--exclude=quick_test.sh' flook-0.7.0.old/smeka/Makefile.install flook.new/smeka/Makefile.install
--- flook-0.7.0.old/smeka/Makefile.install	2017-02-28 08:40:09.000000000 +0100
+++ flook.new/smeka/Makefile.install	2018-02-09 14:45:59.763078077 +0100
@@ -13,16 +13,16 @@
 .NOTPARALLEL: smeka-install-lib
 smeka-install-lib: smeka-install-init-dir
 ifneq ($(LIBRARIES),)
-	install -pm $(_OCT_rwxrxrx) -t $(LIB_DIR) $(LIBRARIES)
+	install -pm $(_OCT_rwxrxrx) -t $(DESTDIR)$(LIB_DIR) $(LIBRARIES)
 endif
 
 
 .PHONY: smeka-install-init-dir
 .NOTPARALLEL: smeka-install-init-dir
 smeka-install-init-dir:
-	-mkdir -p $(INC_DIR)
-	-mkdir -p $(LIB_DIR)
-	-mkdir -p $(BIN_DIR)
+	-mkdir -p $(DESTDIR)$(INC_DIR)
+	-mkdir -p $(DESTDIR)$(LIB_DIR)
+	-mkdir -p $(DESTDIR)$(BIN_DIR)
 
 
 # Issuing uninstall will forcefully remove the installation directory
@@ -31,14 +31,14 @@
 smeka-uninstall-install:
 	@echo ""
 	@echo "The directory tree:"
-	@echo "  $(PREFIX)"
+	@echo "  $(prefix)"
 	@echo "will be deleted!"
 	@echo "You may abort the operation within 2 secs by pressing:"
 	@echo ""
 	@echo "  Ctrl+C or ^C"
 	@echo ""
 	@sleep 2
-	@rm -rf $(PREFIX)
+	@rm -rf $(prefix)
 uninstall: smeka-uninstall-install
 
 
