diff -urN flook-0.7.0.old/Makefile.project flook-0.7.0.new/Makefile.project
--- flook-0.7.0.old/Makefile.project	2019-08-29 18:40:29.671568014 +0200
+++ flook-0.7.0.new/Makefile.project	2019-08-29 18:50:51.584677626 +0200
@@ -51,6 +51,9 @@
  endif
 endif
 
+# Set default install script
+INSTALL ?= install -p
+
 
 # Create custom installation of modules
 .PHONY: install-mod
@@ -60,7 +63,7 @@
 ifneq ($(LIBRARIES),)
 install-mod:
 # This should only pass if one wishes to install the flook library as well
-	install -pm $(_OCT_rwrr) -t $(DESTDIR)$(FMOD_DIR) *.mod
+	$(INSTALL) -m $(_OCT_rwrr) -t $(DESTDIR)$(FMOD_DIR) *.mod
 endif
 
 
diff -urN flook-0.7.0.old/aotus/Makefile.project flook-0.7.0.new/aotus/Makefile.project
--- flook-0.7.0.old/aotus/Makefile.project	2017-02-16 22:12:01.000000000 +0100
+++ flook-0.7.0.new/aotus/Makefile.project	2019-08-29 18:49:52.312843073 +0200
@@ -37,6 +37,9 @@
 EXTDOUBLE ?= 0
 QUADRUPLE ?= 0
 
+# Set default install script
+INSTALL ?= install -p
+
 
 # Create custom installation of modules
 .PHONY: install-mod
@@ -46,7 +49,7 @@
 install-mod:
 ifneq ($(LIBRARIES),)
 # This should only pass if one wishes to install the aotus library as well
-	install -pm $(_OCT_rwrr) -t $(FMOD_DIR) *.mod
+	$(INSTALL) -m $(_OCT_rwrr) -t $(FMOD_DIR) *.mod
 endif
 
 
diff -urN flook-0.7.0.old/aotus/smeka/Makefile.install flook-0.7.0.new/aotus/smeka/Makefile.install
--- flook-0.7.0.old/aotus/smeka/Makefile.install	2017-02-16 22:12:01.000000000 +0100
+++ flook-0.7.0.new/aotus/smeka/Makefile.install	2019-08-29 18:49:34.768884654 +0200
@@ -1,4 +1,5 @@
 _SMEKA_install = 1
+INSTALL ?= install -p
 
 # Basic implementation of installing the different
 # libraries etc.
@@ -13,7 +14,7 @@
 .NOTPARALLEL: smeka-install-lib
 smeka-install-lib: smeka-install-init-dir
 ifneq ($(LIBRARIES),)
-	install -pm $(_OCT_rwxrxrx) -t $(LIB_DIR) $(LIBRARIES)
+	$(INSTALL) -m $(_OCT_rwxrxrx) -t $(LIB_DIR) $(LIBRARIES)
 endif
 
 
diff -urN flook-0.7.0.old/smeka/Makefile.install flook-0.7.0.new/smeka/Makefile.install
--- flook-0.7.0.old/smeka/Makefile.install	2019-08-29 18:40:29.671568014 +0200
+++ flook-0.7.0.new/smeka/Makefile.install	2019-08-29 18:50:10.480796361 +0200
@@ -1,4 +1,5 @@
 _SMEKA_install = 1
+INSTALL ?= install -p
 
 # Basic implementation of installing the different
 # libraries etc.
@@ -13,7 +14,7 @@
 .NOTPARALLEL: smeka-install-lib
 smeka-install-lib: smeka-install-init-dir
 ifneq ($(LIBRARIES),)
-	install -pm $(_OCT_rwxrxrx) -t $(DESTDIR)$(LIB_DIR) $(LIBRARIES)
+	$(INSTALL) -m $(_OCT_rwxrxrx) -t $(DESTDIR)$(LIB_DIR) $(LIBRARIES)
 endif
 
 
