--- SRC/get_perm_c_parmetis.c.orig	2018-02-10 23:52:51.803060282 +0100
+++ SRC/get_perm_c_parmetis.c	2018-02-10 23:53:11.786969214 +0100
@@ -26,6 +26,10 @@ at the top-level directory.
 #include <math.h>
 #include "superlu_dist_config.h"
 #ifdef HAVE_PARMETIS
+#include <stdint.h>
+#include <stdio.h>
+#include "mpi.h"
+#include "ptscotch.h"
 #include "parmetis.h"
 #endif
 #include "superlu_ddefs.h"
--- CMakeLists.txt.oig	2018-02-14 16:04:42.918476506 +0100
+++ CMakeLists.txt	2018-02-14 16:05:42.253613599 +0100
@@ -181,44 +181,7 @@
 endif()
 
 #--------------------- ParMETIS ---------------------
-if (enable_parmetislib)   ## want to use parmetis
-  if (NOT TPL_PARMETIS_LIBRARIES)
-    message(FATAL_ERROR "TPL_PARMETIS_LIBRARIES option should be set for PARMETIS support to be enabled.")
-  endif()
-
-  if (NOT TPL_PARMETIS_INCLUDE_DIRS)
-    message(FATAL_ERROR "TPL_PARMETIS_INCLUDE_DIRS option be set for PARMETIS support to be enabled.")
-  endif()
-  foreach(dir ${TPL_PARMETIS_INCLUDE_DIRS})
-    if (NOT EXISTS ${dir})
-      message(FATAL_ERROR "PARMETIS include directory not found: ${dir}")
-    endif()
-    set(CMAKE_C_FLAGS "-I${dir} ${CMAKE_C_FLAGS}")
-  endforeach()
-
-  message("-- Enabled support for PARMETIS")
-  set(PARMETIS_FOUND TRUE)
-
-  set(PARMETIS_LIB ${TPL_PARMETIS_LIBRARIES})
-  # fix up PARMETIS library names
-  string (REPLACE ";" " " PARMETIS_LIB_STR "${PARMETIS_LIB}")
-  set(PARMETIS_LIB_EXPORT ${PARMETIS_LIB_STR})
-
-else()
-  message("-- Will not link with ParMETIS.")
-endif()
-
-if(enable_parmetislib AND NOT PARMETIS_FOUND)
-  find_package(ParMETIS)
-  if(PARMETIS_FOUND)
-    set(PARMETIS_LIB ParMETIS::ParMETIS)
-    set(TPL_PARMETIS_INCLUDE_DIRS "")
-  endif()
-endif()
-
-if(PARMETIS_FOUND)
-  set(HAVE_PARMETIS TRUE)
-endif()
+set(HAVE_PARMETIS TRUE)
 
 
 ######################################################################
@@ -229,9 +192,6 @@
 
 include_directories(${CMAKE_BINARY_DIR}/SRC) # For superlu_dist_config.h
 include_directories(${CMAKE_SOURCE_DIR}/SRC)
-if (TPL_PARMETIS_INCLUDE_DIRS)
-  include_directories(${TPL_PARMETIS_INCLUDE_DIRS})  ## parmetis
-endif ()
 include_directories(${MPI_C_INCLUDE_PATH})
 
 ######################################################################
