--- Makefile.orig	2020-02-21 11:03:41.230183064 +0000
+++ Makefile	2020-02-22 11:21:07.055057712 +0000
@@ -10,6 +10,18 @@ default: wannier post
 
 all: wannier lib post w90chk2chk w90pov w90vdw
 
+PREFIX ?=
+
+install: default
+	install -d $(DESTDIR)$(PREFIX)/bin/
+	for x in wannier90.x postw90.x w90chk2chk.x w90spn2spn.x ; do \
+		if [ -f "$$x" ]; then install -m755 "$$x" "$(DESTDIR)$(PREFIX)/bin/$$x"; fi; \
+	done
+	if [ -f "utility/w90pov/w90pov" ]; then install -m755 "utility/w90pov/w90pov" "$(DESTDIR)$(PREFIX)/bin/w90pov"; fi;
+	if [ -f "utility/w90vdw/w90vdw.x" ]; then install -m755 "utility/w90vdw/w90vdw.x" "$(DESTDIR)$(PREFIX)/bin/w90vdw.x"; fi;
+	install -d $(DESTDIR)$(PREFIX)/lib/
+	if [ -f "libwannier.a" ]; then install -m644 "libwannier.a" "$(DESTDIR)$(PREFIX)/lib/libwannier.a"; fi;
+
 doc: thedoc
 
 serialobjs: objdir
--- /dev/null	2020-02-22 08:25:20.608170301 +0000
+++ make.inc	2020-02-22 10:59:55.824980033 +0000
@@ -0,0 +1,15 @@
+
+#=====================================================
+# For Linux with intel version 11/12 on 64bit machines
+#=====================================================
+F90 ?= gfortran
+COMMS?=mpi
+MPIF90?=mpif90
+FCOPTS?=-O3
+LDOPTS?=-O3
+
+#=======================
+# ATLAS Blas and LAPACK
+#=======================
+LIBS ?= -lopenblas
+
